code/datums/mergers/_merger.dm

```diff
@@ -0,0 +1,132 @@
+#define MERGERS_DEBUG FALSE
+
+/// A datum that tracks a type or types of objects in a cluster
+/datum/merger
+	/// The unique ID for this merger datum, adjacent merg groups with the same id will combine
+	var/id
+	/// The types allowed to be in this merge group
+	var/list/merged_typecache
+	/// Optional proc to call on potential members, return true to allow merge
+	var/attempt_merge_proc
+
+	/// The arbitrary "owner" member of the merge group
+	var/atom/origin
+	/// A list of all members in the group
+	var/list/members = list()
+
+#if MERGERS_DEBUG
```

Comment:
is this defined anywhere? maybe throw it in compile_options

Reply:
It's defined within this file and undefined at the end

Reply:
ah scuse me